lib/src/client_messages.dart

```diff
@@ -102,37 +107,40 @@ class StartupMessage extends ClientMessage {
 }
 
 class QueryMessage extends ClientMessage {
-  final UTF8BackedString _queryString;
+  final EncodedString _queryString;
 
-  QueryMessage(String queryString)
-      : _queryString = UTF8BackedString(queryString);
+  QueryMessage(String queryString, Encoding encoding)
```

Comment:
Technically all of these are breaking changes, although in practice they may not be relevant for most users. Maybe `encoding` should a named nullable parameter in all of the messages, defaulting to `utf8` if not set?

Reply:
I put it as a mandatory parameter because I think it's easier to manage code maintenance, because as mandatory I already see in the IDE all the places I have to change so I don't forget to pass on the correct instance of the Codec that comes from PostgreSQLConnection, and also because I think it is more centralized to see the PostgreSQLConnection codec instance, there is only one place to instantiate the codec, in addition I save having to check if the value is null to get the utf8 instance